src/Core/MySQL/MySQLGtid.cpp

```diff
@@ -0,0 +1,178 @@
+#include "MySQLGtid.h"
+
+#include <boost/algorithm/string.hpp>
+
+namespace DB
+{
+namespace ErrorCodes
+{
+    extern const int UNKNOWN_EXCEPTION;
+}
+
+void GTIDSet::tryMerge(size_t i)
+{
+    if ((i + 1) >= intervals.size())
+        return;
+
+    if (intervals[i].end != intervals[i + 1].start)
+        return;
+    intervals[i].end = intervals[i + 1].end;
+    intervals.erase(intervals.begin() + i + 1, intervals.begin() + i + 1 + 1);
+}
+
+void GTIDSets::parse(const String gtid_format)
+{
+    if (gtid_format.empty())
+    {
+        return;
+    }
+
+    std::vector<String> gtid_sets;
+    boost::split(gtid_sets, gtid_format, [](char c) { return c == ','; });
```

Comment:
GTID sets format is only used in the `.metadata` file, it parsed and to string by MySQLGtid module only, it can't be set manually.

Actually, there is no spec on GTID sets format even in MySQL offical docs.
MySQL adds ', (whitesapce)' after comma here to give a good look for human, but it compatible with non-whitesapce mode(',').

Reply:
Updated:
Added GTID sets  ` ", " `(comma and whitespace) compatibility:
```
2c5adab4-d64a-11e5-82df-ac162d72dac0:1-3, 9f58c169-d121-11e7-835b-ac162db9c048:1-5
```
and 
```
2c5adab4-d64a-11e5-82df-ac162d72dac0:1-3,9f58c169-d121-11e7-835b-ac162db9c048:1-5
```
both works.